### DISCUSSION TITLE
Is there a way to allow users to sign up for a trial before downloading the extension?

### DISCUSSION THREAD

**Discussion by MattCorwin** [Glench/ExtPay#182] [Q&A]
I'd like to let users click 'Start a Trial' and sign up from my landing page before being routed to download the extension, so that when they download the extension the trial/paid features are already enabled and they don't have to go through any extra steps. Has anyone implemented this? Is it even possible?

**Comment by burkybang** [comment]
The user's information gets saved in the extension's storage. The storage container doesn't exist until after the user installs the extension.

Also, to start a free trial, the user needs to click a link in an email from ExtensionPay that opens an ExtensionPay page. Your extension is intended to listen for an event on that page. You wouldn't have any control over what happens on that page unless the extension is installed.

I don't see any way to do what you're asking with how ExtensionPay is currently 😞
